### PR TITLE
Safer Updater

### PR DESCRIPTION
**Pull request by taooceros**
might close https://github.com/Flow-Launcher/Flow.Launcher/issues/1504

Note:
I am not so sure why the autoupdater may crash flow itself...I guess it is because the Elapsed callback is triggered from the main thread, but I cannot find evidence to support it, because based on document, this happens only when SynchronizingObject is set which is not we have done.

Though, I have suppressed the error of general exception from updater, so now at least it will be safer to use the autoupdater.

### CODE REVIEW

**Comment by jjw24** [comment]
Done a ln update test to ensure working as expected?

File: Flow.Launcher/App.xaml.cs [hunk 1]
```diff
@@ -84,7 +85,7 @@ private async void OnStartupAsync(object sender, StartupEventArgs e)
 
                 Current.MainWindow = window;
                 Current.MainWindow.Title = Constant.FlowLauncher;
-                
+
                 HotKeyMapper.Initialize(_mainVM);
 
                 // happlebao todo temp fix for instance code logic
```
Comment (jjw24) [inline Flow.Launcher/App.xaml.cs:91]:
```suggestion
                // todo temp fix for instance code logic
```

**Comment by jjw24** [comment]
Can you please fix spelling errors 'Uodate', 'settigns'.

File: Flow.Launcher.Core/Updater.cs [hunk 2]
```diff
@@ -88,9 +88,13 @@ public async Task UpdateAppAsync(IPublicAPI api, bool silentUpdate = true)
                     UpdateManager.RestartApp(Constant.ApplicationFileName);
                 }
             }
-            catch (Exception e) when (e is HttpRequestException or WebException or SocketException || e.InnerException is TimeoutException)
+            catch (Exception e)
             {
-                Log.Exception($"|Updater.UpdateApp|Check your connection and proxy settings to github-cloud.s3.amazonaws.com.", e);
+                if ((e is HttpRequestException or WebException or SocketException || e.InnerException is TimeoutException))
+                    Log.Exception($"|Updater.UpdateApp|Check your connection and proxy settings to github-cloud.s3.amazonaws.com.", e);
```
Comment (jjw24) [inline Flow.Launcher.Core/Updater.cs:94]: Looking at this, I wonder if we need to notify users when these type of errors occur?

File: Flow.Launcher/App.xaml.cs [hunk 3]
```diff
@@ -130,20 +131,17 @@ private void AutoStartup()
...
-                    timer.Start();
-
-                    // check updates on startup
+                    // check update every 5 hours
+                    var timer = new PeriodicTimer(TimeSpan.FromHours(5));
                     await _updater.UpdateAppAsync(API);
+
+                    while (await timer.WaitForNextTickAsync())
+                        // check updates on startup
+                        await _updater.UpdateAppAsync(API);
```
Comment (jjw24) [inline Flow.Launcher/App.xaml.cs:144]: is this a double up of `await _updater.UpdateAppAsync(API);`? Doesnt it run once timer is instantiated?

**Comment by jjw24** [comment]
@taooceros have you tested updating with this change?

File: Flow.Launcher.Core/Updater.cs (hunk 2 above)
Comment (taooceros) [reply Flow.Launcher.Core/Updater.cs:94]: I think we did? In the later part when silentupdate is false

**Review by jjw24 — APPROVED** [review]

**Pull request merged**